tabular/src/autogluon/tabular/models/abstract/abstract_model.py

```diff
@@ -601,7 +614,10 @@ def hyperparameter_tune(self, X_train, y_train, X_val, y_val, scheduler_options,
         scheduler.run()
         scheduler.join_jobs()
 
-        return self._get_hpo_results(scheduler=scheduler, scheduler_options=scheduler_options, time_start=time_start)
+        hpo_models, hpo_model_performances, hpo_results = self._get_hpo_results(scheduler=scheduler, scheduler_options=scheduler_options, time_start=time_start)
+        if 'report_probs' in kwargs:
+            hpo_results['ensemble'] = scheduler.ensemble
```

Comment:
You are assuming here that if report_probs is given, the scheduler is of a very specific type, so it has ensemble member. This is a little reckless, at least the whole thing has to be documented somewhere.

Reply:
This is also for hpo over ensemble. report_probs is a flag, it doesn't actually contain any probability data. I have added another check hasattr(scheduler, "ensemble") to be safe. Also added the comment to document this change.